src/main/java/com/mraof/minestuck/tileentity/machine/SendificatorTileEntity.java

```diff
@@ -0,0 +1,232 @@
+import net.minecraftforge.items.ItemStackHandler;
+import net.minecraftforge.items.wrapper.RangedWrapper;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class SendificatorTileEntity extends MachineProcessTileEntity implements INamedContainerProvider
+{
+	public static final RunType TYPE = RunType.BUTTON_OVERRIDE;
+	public static final String TITLE = "container.minestuck.sendificator";
+	public static final int DEFAULT_MAX_PROGRESS = 0;
+	
+	private BlockPos destBlockPos;
+	
+	private final IntReferenceHolder fuelHolder = new IntReferenceHolder()
+	{
+		@Override
+		public int get()
+		{
+			return fuel;
+		}
+		
+		@Override
+		public void set(int value)
+		{
+			fuel = (short) value;
+		}
+	};
+	
+	private short fuel = 0;
+	private static final short maxFuel = 128;
+	
+	public SendificatorTileEntity()
+	{
+		super(MSTileEntityTypes.SENDIFICATOR.get());
+		maxProgress = DEFAULT_MAX_PROGRESS;
+	}
+	
+	public BlockPos getDestinationBlockPos()
+	{
+		if(destBlockPos == null)
+			destBlockPos = new BlockPos(0, 0, 0);
+		return destBlockPos;
+	}
+	
+	public void setDestinationBlockPos(BlockPos destinationPosIn)
+	{
+		this.destBlockPos = destinationPosIn;
+	}
+	
+	@Override
+	public ITextComponent getDisplayName()
+	{
+		return new TranslationTextComponent(TITLE);
+	}
+	
+	@Override
+	public void load(BlockState state, CompoundNBT compound)
+	{
+		super.load(state, compound);
+		
+		int destX = compound.getInt("destX");
+		int destY = compound.getInt("destY");
+		int destZ = compound.getInt("destZ");
+		this.destBlockPos = new BlockPos(destX, destY, destZ);
+		
+		fuel = compound.getShort("fuel");
+	}
+	
+	@Override
+	public CompoundNBT save(CompoundNBT compound)
+	{
+		super.save(compound);
+		
+		getDestinationBlockPos();
+		compound.putInt("destX", destBlockPos.getX());
+		compound.putInt("destY", destBlockPos.getY());
+		compound.putInt("destZ", destBlockPos.getZ());
+		
+		compound.putShort("fuel", fuel);
+		
+		return compound;
+	}
+	
+	@Override
+	protected ItemStackHandler createItemHandler()
+	{
+		return new MachineProcessTileEntity.CustomHandler(3, (index, stack) -> index == 1 ? ExtraForgeTags.Items.URANIUM_CHUNKS.contains(stack.getItem()) : index != 2);
+	}
+	
+	@Override
+	public MachineProcessTileEntity.RunType getRunType()
+	{
+		return TYPE;
+	}
+	
+	@Override
+	public boolean contentsValid()
+	{
+		if(level.hasNeighborSignal(this.getBlockPos()))
+		{
+			return false;
+		}
+		
+		ItemStack fuel = itemHandler.getStackInSlot(1);
+		ItemStack input = itemHandler.getStackInSlot(0);
+		return getFuel() <= getMaxFuel() - 32 && ExtraForgeTags.Items.URANIUM_CHUNKS.contains(fuel.getItem()) || !input.isEmpty();
+	}
+	
+	/**
+	 * With the given container possessing tile entity system our mod uses, this is the function that connects to the GoButton found in it's screen(SendificatorScreen in this example)
+	 */
+	@Override
+	public void processContents()
+	{
+		if(getFuel() <= getMaxFuel() - 32)
```

Comment:
im guessing 32 is the fuel cost, it would be cleaner to have it as a const for future use imo

Reply:
Agreed, will change it to have that, along with the cookalyzer since its what this was based off of